release/models/network-instance/openconfig-programming-errors.yang

```diff
@@ -0,0 +1,240 @@
+
+            leaf prefix {
+              type leafref {
+                path "../state/prefix";
+              }
+              description
+                "Reference to the prefix that keys the failed list.";
+            }
+
+            container state {
+              description
+                "Operational state parameters relating to a failed programming
+                operation.";
+              uses route-state-common;
+            }
+          }
+        }
+
+        container stale-routes {
+          config false;
+          description
+            "Surrounding container for the list of routes that are currently in
+            a stale state.";
+
+          list stale {
+            key "prefix";
+
+            description
+              "A prefix that is currently installed in hardware, but a subsequent
+              operation to update its programming failed - such that the entry in
+              hardware is stale.";
+
+            leaf prefix {
+              type leafref {
+                path "../state/prefix";
+              }
+              description
+                "Reference to the prefix that keys the stale list.";
+            }
+
+            container state {
+              description
+                "Operational state parameters relating to a stale route.";
+              uses route-state-common;
+            }
+          }
+        }
+
+        container drop-routes {
+          config false;
+          description
+            "Surrounding container for the list of routes that are currently in
+            a drop state.";
+
+          list drop {
+            key "prefix";
+
+            description
+              "A prefix that is currently installed in hardware but with an explicit
+              instruction that it should discard packets that are destined towards
+              it. This list includes both intentional and un-intentional
+              drop routes.";
+
+            leaf prefix {
+              type leafref {
+                path "../state/prefix";
+              }
+              description
+                "Reference to the prefix that keys the drop list.";
+            }
+
+            container state {
+              description
+                "Operational state parameters relating to a drop route.";
+              uses route-state-common;
+            }
+          }
+        }
+      }
+    }
+  }
+
+ grouping route-state-common {
+    description
+      "Common parameters that correspond to a particular route type.";
+
+    leaf prefix {
+      type oc-inet:ip-prefix;
+      description
+        "The IPv4 or IPv6 prefix that the route state corresponds to.";
+    }
+
+    leaf time {
+      type oc-types:timeticks64;
+      description
+        "Represents the time the route programming state change was detected by the
+        monitoring subsystem expressed as nanoseconds since the Unix epoch.";
+    }
+
+    leaf dest-component {
```

Comment:
dest-component should be able to support multiple components (ie, modular chassis).

Reply:
Are you suggesting to convert this into a `leaf-list`?

Reply:
Changed it to `leaf-list` in latest diff.